### ISSUE TITLE
Suggestion - A method to disable Toolbox on PocketMine server side

### ISSUE THREAD

**Issue by Leo3418** [MCMrARM/toolbox-issue-tracker#47]
Toolbox has become a popular cheating tool on many PocketMine servers because it provides functionalities like X-ray and faster movement. We don't want our players use it since it makes the game unfair.
Is there any way to block or disable these functions from server side, for example a PocketMine plugin that defenses Toolbox?
Thank you very much!

**Comment by slice** [comment]
Although I am not a developer of Toolbox, I agree that this is **very** important. @MCMrARM 

Also @Leo3418, do you mean players using BlockLauncher with Toolbox? Or just using the Toolbox launcher? Both provide similar functionality, but they still differ.

**Comment by Leo3418** [comment]
@sliceofcode thanks. The player who reported possibility of cheating to us only used Toolbox. After that I tested Toolbox without BlockLauncher, X-ray and changing game speed work.

**Comment by ghost** [comment]
I think this will be hard, BlockLauncher disables the mods but there's some mods that Re-Enables ...

**Comment by Blakintosh** [comment]
Nah, the variables that determine whether the scripts are enabled are private final java variables, which cannot be accessed from a ModPE script AFAIK. But BL does not disable the UIs of the mod, so e.g a button that appears in the main menu will stay there when entering a server and can then be used to activate a mod menu, such as Ghost Hack.

**Comment by TheBigHacker0100** [comment]
No it won't, update fixes this, (I don't know if it is in the latest update but last time I used BlockLauncher it did not work, Ghost Hack got disabled) there is some sendFakeCallback that bypasses the disabling

**Comment by Johnmacrocraft** [comment]
@Smasher535 @Black-DeathPE @TheBigHacker0100 BlockLauncher got patched of that few months ago, but we can actually bypass it using setLevel... and is... (Hiding name because it can be abused other users); it's public class, not private. (Yea, actually it's kind of easter egg feature)

**Comment by AutoGrind** [comment]
Leo don't be lazy. Make plugins to block certain things from happening or make troll blocks and put some time and effort into catching cheaters on your own who use X-ray. If you can disable toolbox server-side someone else will make another launcher and toolbox will lose popularity.

**Comment by Leo3418** [comment]
@AutoGrind Thanks for the suggestion. I just hope that I have sufficient skill to make such plugin you described.

**Comment by AutoGrind** [comment]
@leo3418 Thanks for taking that the way I meant it and not as an abusive comment to you. It's kind of hard to expect one to give you the godly ability to disable something on the users end when it will be bypassed eventually with the same launcher or by many others to come. Hacking is sometimes used as a form of free speech to tell server owners/developers to get off their butts and keep developing lol. 
So as an example: if you use an anti fly plugin it will stop most who are flying regardless of the launcher they use. But if you get one launcher to give you the ability to stop flight they user will just use another to be able to fly in your server. So it's like wasted time and lost sleep for you compared to using anti fly plugins.
